### PR TITLE
Fix typos and small mistakes in Chap4

### PR DESCRIPTION
**Pull request by mikakruschel**
Hi, these are the things I noticed while reading chapter 4 and fixed:

- ~swapped `.` with `”`~

- changed number of lines for comments in some cases
- changed "use" to "using" in line 412

> I'm ~use~ using a solution to Exercise 4.1 and assume a run() method that handles all of the particle's functionality.

- changed "ArrayList" to "Array"
- shortened `isDead` after it was already explained in line 1676
- added `this.` in line 1826

TODO: In line 1592 is an "inline" comment, which is not handled by magicbook

### CODE REVIEW

**Comment by mikakruschel** [comment]
I just read that commas inside quotes is common in American English so I will undo that.

File: content/04_particles.html [hunk 1]
```diff
@@ -574,9 +574,9 @@ <h2>4.3 The Array</h2>
       doesn’t cause the program to crash (as it does with adding), the problem
       is almost more insidious in that it leaves no evidence. To discover the
       flaw I must first establish an important fact. When an object is
-      removed from the array with<code>splice()</code>, all elements are shifted one spot
+      removed from the array with <code>splice()</code>, all elements are shifted one spot
       to the left. Note the diagram below where particle C (index 2) is removed.
-      particles A and B keep the same index, while particles D and E shift from
+      Particles A and B keep the same index, while particles D and E shift from
```
Comment (shiffman) [inline content/04_particles.html]: I think I will use the lower case p when referring to particles generally and capital `Particle` when talking about the Particle class.

```suggestion
      particles A and B keep the same index, while particles D and E shift from
```

File: content/04_particles.html [hunk 2]
```diff
@@ -932,7 +932,7 @@ <h5>Example 4.4: System of systems</h5>
     </div>
 
     <pre data-code-language="javascript" data-type="programlisting" class="codesplit">
-//{!1} This time, the type of thing we are putting in the ArrayList is a ParticleSystem itself!
+//{!1} This time, the type of thing we are putting in the Array is a ParticleSystem itself!
```
Comment (shiffman) [inline content/04_particles.html]:
```suggestion
//{!1} This time, the type of thing we are putting in the array is a particle system itself!
```

**Review by shiffman — APPROVED** [review]
Sorry I am only just now getting to this, thanks for these wonderful fixes!

**Pull request merged**